services/twilioService.js

```diff
@@ -0,0 +1,225 @@
+import { Client as TwilioChatClient } from 'twilio-chat';
+import { AsyncStorage } from 'react-native';
+import { CONFIG } from '../config.js';
+import MessageService from './messageService.js';
+
+const API_URL = CONFIG.ENV === 'PROD' ? CONFIG.API_URL : CONFIG.TEST_API_URL;
+
+class TwilioService {
+  constructor(hmToken, contacts, messagesCallback) {
+    this.hmToken = hmToken;
+    this.id = `${JSON.parse(hmToken)._id}`;
+
+    this.chatClient = null;
+    this.messageService = new MessageService(messagesCallback);
+
+    // Map user_id to chat channel object
+    this.channels = {};
+    this.contacts = [contacts];
+
+    this.startup(contacts);
+  }
+
+  async startup(contacts) {
+    const success = await this.loadTwilioClient();
+    if (success) {
+      // Initialize all channels
+      const doneInitChannels = await this.initAllChannels(contacts);
+      if (doneInitChannels) {
+        await this.getAllMessages();
+      } else {
+        // TODO: something bad happened
+      }
+    } else {
+      // TODO: Add failure logging
+    }
+  }
+
+  async getAllMessages() {
+    await this.contacts.map(contact => this.updateMessages(contact));
+  }
+
+  async sendMessage(contact, message) {
+    try {
+      const c = this.channels[contact];
+      c.sendMessage(message);
+    } catch (e) {
+      // TODO: add sentry logging
+    }
+  }
+
+  async initAllChannels(contacts) {
+    const success = await Promise.all(contacts.map(contact => this.initSingleChannel(contact)));
```

Comment:
This returns an array of booleans, you probably want to check them all:

`return success.every(val => val)`

Reply:
Thanks, fixed in next push.